### DISCUSSION TITLE
Issue Azure SSO

### DISCUSSION THREAD

**Discussion by lucienn3** [kimai/kimai#2331] [Questions & Answers] [answered]
Hey I was trying to follow your SAML steps for Azure SSO, but it is not working.

I tried to make a comment in https://github.com/kevinpapst/kimai2/issues/1975 but the issue is already closed.

Hey,

I was trying to configure this as well.

When I try your provided configuration:
`connection:
            idp:
                entityId: '*********'
                singleSignOnService:
                    url: 'https://login.microsoftonline.com/*********/saml2'
                    binding: 'urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect'
                x509cert: '-----BEGIN PRIVATE KEY-----*********-----END PRIVATE KEY-----'
            sp:
                entityId: 'https://timetracking.*********/auth/saml/metadata'
                assertionConsumerService:
                    url: 'https://timetracking.*********/auth/saml/acs'
                    binding: 'urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST'
                singleLogoutService:
                    url: 'https://timetracking.*********/auth/saml/logout'
                    binding: 'urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect'`

If I put https://timetracking.*********/auth/saml/metadata in the entityId then I get the following issue:
`AADSTS700016: Application with identifier 'https://timetracking.*********/auth/saml/metadata' was not found in the directory '*********'. This can happen if the application has not been installed by the administrator of the tenant or consented to by any user in the tenant. You may have sent your authentication request to the wrong tenant.`

And if I use the actual application ID I get the following:
`AADSTS50011: The reply URL specified in the request does not match the reply URLs configured for the application: '*********'.`

Could you please help me?

Best Regards,
Lucien

**Comment by kevinpapst** [comment]
Sorry, I don't have an Azure account for testing. Maybe @Gyrfalc0n can help ... and maybe you need to play with the configurations on Kimai and Azure side. Sorry.

**Reply by lucienn3** [reply]
Would you maybe have some time to troubleshoot this with me? I would be happy to work with your environment + my Azure to troubleshoot this and then enhance the documentation as a result. I can host Teams meeting and share application details to then test this. And then write a piece of documentation for your wiki.

You can contact me directly using my LinkedIn:
https://www.linkedin.com/in/lucien-sch%C3%BCnke-70199654/

**Reply by Gyrfalc0n** [reply]
Hello,

Normally in your Azure, you have a SAML configuration test panel. You can directly indicate the error in the corresponding field. Try to indicate your error.

The test panel of finds in your application/single sign on below.

![image](https://user-images.githubusercontent.com/46728024/107291551-a73f0400-6a68-11eb-9b96-cb204a0b34b7.png)
![image](https://user-images.githubusercontent.com/46728024/107291598-b7ef7a00-6a68-11eb-9b99-480801546e71.png)

**Reply by lucienn3** [reply]
@Gyrfalc0n 
Thanks again!

I would be interested, what do you think of the news steps? https://github.com/lucienn3/www.kimai.org/blob/7a0fbd046b1f93d2e07ee9d4d524024c25e2f9a1/_documentation/azure.md

**Comment by lucienn3** [comment]
Hey, good news! I finally found how to access this view... :)

I will test configure and test it and then(if it works) let's make it more clear how to get to this :)

**Comment by lucienn3** [comment]
Hey,

Good News!

It works for me :)

I will create a better documentation piece tomorrow, then retest it with a new environment + application to test the steps and verify. After that, I will provide them here. If possible it would be great if you guys, could recheck the steps and let me know if it also works for you :)?

Regards,
Lucien

**Comment by lucienn3** [comment, marked as answer]
@kevinpapst Hey, I have checked your website, but cannot see how to edit the documentation.

What would be the best way to provide to you a full set of steps for https://www.kimai.org/documentation/saml.html ? :)

**Reply by kevinpapst** [reply]
Thanks that you want to share your findings!

You have missed that link I guess:
<img width="1271" alt="Bildschirmfoto 2021-02-11 um 10 57 56" src="https://user-images.githubusercontent.com/533162/107622409-16c71600-6c58-11eb-9731-f64041f152fe.png">

Its pointing to the docs repository at: https://github.com/kimai/www.kimai.org/blob/master/_documentation/saml.md

**Reply by lucienn3** [reply]
Thanks :)

I uploaded it...
Pull Request: https://github.com/kimai/www.kimai.org/pull/89
To view the new doc: https://github.com/lucienn3/www.kimai.org/blob/7a0fbd046b1f93d2e07ee9d4d524024c25e2f9a1/_documentation/azure.md

Please let me know what you think.

PS: Can you please download the pictures from my server and put them on GitHub instead of using my own picture URLs? I just didn't know how to make it work :D in Github.
